src/lib.rs

```diff
@@ -191,15 +186,6 @@ pub fn gather_telemetry_from_file(file: File, tx: Sender<TelemetryChannelType>)
                     match parse_telemetry_message(&buffer) {
                         // It worked! Let's extract the message and replace the buffer with the rest of the bytes
                         Ok((rest, message)) => {
-                            match message {
```

Comment:
Unfortunately if you move this to the binary this will break running the control GUI from a file. Maybe we just need to add a parameter to the public function to enable/disable time "simulation"?

Reply:
IIUC `gather_telemetry_from_file` serve 2 purposes. Gather telemetry from the file, and send it to channel with time simulation. As you suggest, I will add a parameter to manage the second purpose.